src/python/WMCore/WMSpec/Steps/Executors/DQMUpload.py

```diff
@@ -293,3 +309,229 @@ def upload(self, url, args, filename):
+        opener.add_handler(handler)
+
+        # setup the request object
+        url = decodeBytesToUnicode(registerURL)
+        datareq = Request(url)
+        datareq.add_header('Accept-encoding', 'gzip')
+        datareq.add_header('User-agent', ident)
+        datareq.data = encodeUnicodeToBytes(json.dumps(args))
+
+        if 'https://' in url:
+            result = opener.open(datareq)
+        else:
+            opener.add_handler(ProxyHandler({}))
+            result = opener.open(datareq)
+
+        return result
+    
+    def _registerAnalysisFile(self, stepName, analysisFile):
+        """
+        Register a file to new DQM GUi
+        - https://github.com/cms-DQM/dqmgui
+        HTTP request body:
+            [{"dataset": "/a/b/c", "run": "123456", "lumi": "0", "file": "/eos/cms/store/group/comm_dqm/DQMGUI_data/location/file.root", "fileformat": 1}]
+        Set lumis to 0, as this is reproducing current per Run based root files
+        - https://github.com/dmwm/WMCore/issues/10287#issuecomment-1052558061
+        DQM root files produced by Harvesting prosessing are type 1 (plain ROOT)
+        - https://github.com/cms-DQM/dqmgui#file-formats
+        """
+        # Get task description
+        job = Bootstrap.loadJobDefinition()
+        jobBag = job.getBaggage()
+        datasetName = job.get('inputDataset', None)
+
+        # Get runNumber, depending on the harvesting job mode (byRun vs multiRun)
+        multiRun = getattr(jobBag, "multiRun", False)
+        forceRunNumber = getattr(jobBag, "forceRunNumber", 999999)
+        if multiRun:
+            runNumber = forceRunNumber
+        else:
+            try:
+                runNumber = list(job['mask']['runAndLumis'].keys())[0]
+            except Exception as ex:
+                msg = "Error while retrieving run number from job definition:\n %s" % str(ex)
+                raise WMExecutionFailure(70320, "DQMUploadFailure", msg)    
+
+        args = {}
+        args['file'] = self.registerEOSPrefix + \
+                self.getAnalysisFileLFN(analysisFile)
+        args['dataset'] = datasetName
+        args['run'] = runNumber
+        args['lumi'] = 0
+        args['fileformat'] = 1
+
+        msg = "HTTP Upload is about to start:\n"
+        msg += " => URL: %s\n" % self.registerURL
+        msg += " => Filename: %s\n" % args['file']
+        logging.info(msg)
+
+        try:
+            logException = True
+            for registerURL in self.registerURL.split(';'):
+                result = self._register(registerURL, [args])
+                msg = '  Status code: %s\n' % result.getcode()
+                if result.getcode() >= 400 or result.getcode() is None:
+                    logException = False
+                    raise Exception(msg)
+                else:
+                    msg = 'HTTP POST to register url finished succesfully with response:\n' + msg
+                    logging.info(msg)
+        except Exception as ex:
```

Comment:
Given that we have a retry logic for the root file stage out, I think we should recycle the same logic (or something similar) to this POST requests. If we get this far, then we should try as hard as we can not to fail this job.

Reply:
That's fair. I will implement some retrial feature for this method